llvm/test/MC/RISCV/rv32p-valid.s

```diff
@@ -0,0 +1,1440 @@
+
+nclipri t3, s0
+// CHECK-ASM: nclipri	t3, s0                          # encoding: [0x1b,0xce,0x08,0x74]
+
+pnsrl.bs t3, s0, a4
+// CHECK-ASM: pnsrl.bs	t3, s0, a4                      # encoding: [0x1b,0xce,0xe8,0x08]
+
+pnsrl.hs a2, t1, a4
+// CHECK-ASM: pnsrl.hs	a2, t1, a4                      # encoding: [0x1b,0xc6,0xe6,0x0a]
+
+nsrl a2, a2, a0
+// CHECK-ASM: nsrl	a2, a2, a0                      # encoding: [0x1b,0xc6,0xac,0x0e]
+
+pnclipu.bs a4, t5, a2
+// CHECK-ASM: pnclipu.bs	a4, t5, a2              # encoding: [0x1b,0xc7,0xce,0x28]
+
+pnclipu.hs t1, a2, a4
+// CHECK-ASM: pnclipu.hs	t1, a2, a4              # encoding: [0x1b,0xc3,0xec,0x2a]
+
+nclipu t5, s2, t1
+// CHECK-ASM: nclipu	t5, s2, t1                      # encoding: [0x1b,0xcf,0x62,0x2e]
+
+pnclipru.bs t5, s2, s2
+// CHECK-ASM: pnclipru.bs	t5, s2, s2              # encoding: [0x1b,0xcf,0x22,0x39]
+
+pnclipru.hs t5, s2, a0
+// CHECK-ASM: pnclipru.hs	t5, s2, a0              # encoding: [0x1b,0xcf,0xa2,0x3a]
+
+nclipru a4, t5, t5
+// CHECK-ASM: nclipru	a4, t5, t5                      # encoding: [0x1b,0xc7,0xee,0x3f]
+
+pnsra.bs a4, t1, a4
+// CHECK-ASM: pnsra.bs	a4, t1, a4                      # encoding: [0x1b,0xc7,0xe6,0x48]
+
+pnsra.hs s0, s2, t3
+// CHECK-ASM: pnsra.hs	s0, s2, t3                      # encoding: [0x1b,0xc4,0xc2,0x4b]
+
+nsra t1, s0, a4
+// CHECK-ASM: nsra	t1, s0, a4                      # encoding: [0x1b,0xc3,0xe8,0x4e]
+
+pnsrar.bs a2, s0, a4
+// CHECK-ASM: pnsrar.bs	a2, s0, a4              # encoding: [0x1b,0xc6,0xe8,0x58]
+
+pnsrar.hs s0, a4, a0
+// CHECK-ASM: pnsrar.hs	s0, a4, a0              # encoding: [0x1b,0xc4,0xae,0x5a]
+
+nsrar a4, a4, s0
+// CHECK-ASM: nsrar	a4, a4, s0                      # encoding: [0x1b,0xc7,0x8e,0x5e]
+
+pnclip.bs t1, t5, t3
+// CHECK-ASM: pnclip.bs	t1, t5, t3              # encoding: [0x1b,0xc3,0xce,0x69]
+
+pnclip.hs a0, a2, a0
+// CHECK-ASM: pnclip.hs	a0, a2, a0              # encoding: [0x1b,0xc5,0xac,0x6a]
+
+nclip t3, t5, t3
+// CHECK-ASM: nclip	t3, t5, t3                      # encoding: [0x1b,0xce,0xce,0x6f]
+
+pnclipr.bs t1, a2, a0
+// CHECK-ASM: pnclipr.bs	t1, a2, a0              # encoding: [0x1b,0xc3,0xac,0x78]
+
+pnclipr.hs a4, s2, t3
+// CHECK-ASM: pnclipr.hs	a4, s2, t3              # encoding: [0x1b,0xc7,0xc2,0x7b]
+
+nclipr t1, t5, a2
+// CHECK-ASM: nclipr	t1, t5, a2                      # encoding: [0x1b,0xc3,0xce,0x7e]
+
+pslli.db a0, s2
+// CHECK-ASM: pslli.db	a0, s2                          # encoding: [0x1b,0x4a,0x82,0x00]
+
+pslli.dh t3, t1
+// CHECK-ASM: pslli.dh	t3, t1                          # encoding: [0x1b,0x4c,0x06,0x01]
+
+pslli.dw a4, t3
+// CHECK-ASM: pslli.dw	a4, t3                          # encoding: [0x1b,0x4e,0x0c,0x02]
+
+psslai.dh t1, a4
+// CHECK-ASM: psslai.dh	t1, a4                  # encoding: [0x1b,0x46,0x0e,0x51]
+
+psslai.dw a0, t3
+// CHECK-ASM: psslai.dw	a0, t3                  # encoding: [0x1b,0x4a,0x0c,0x52]
+
+psext.dh.b t1, t5
```

Comment:
If I disassembled this correctly, the `rdp` field has a value of 0b0110. `t1` is register x6, but if we're supposed to encode pairs, there's a missing divide by 2.

I expect if you test disassembly you'll see that registers x16-x31 don't round trip correctly.